test/10_cpp_standards/cpp11/setup.py

```diff
@@ -0,0 +1,9 @@
+import os, sys
```

Comment:
Can we somehow merge all of these three standards into one `setup.py` and one `spam.cpp` file?

We could access an new environment variable `CIBW_TEST_CPP_STANDARD` being `11`, `14`, `17` (or `20`, soon :-) ), and forward this one to `extra_compile_args`.

We could then also pass that as preprocessor variable to `define_macros` and have a single `spam.cpp` file with some preprocessor `#if`? Ór, we could use `__cplusplus` to detect the C++ standard used in the compilation (https://github.com/pybind/pybind11/blob/deb3cb238a9f299d7c57f810165e90a1b14b3e6f/include/pybind11/detail/common.h#L30-L46)

Reply:
Just an idea, though. It would reduce the amount of code we need to maintain, but I don't know how clean the result will look.

Reply:
This will reduce amount of code, but increase complexity, so may make harder to debug. I'm not sure what will be better.

Reply:
I'm not sure it will be so much more complex. Maybe we can give it a try and see?

Reply:
(I can try, if you are OK with that?)

Reply:
control with environment variables?

Reply:
it looks simple, so I push this change.

Reply:
Amazing, that's exactly what I was thinking of! Thanks for trying! -)